class-tour.php

```diff
@@ -44,10 +44,28 @@ public static function enqueue_scripts() {
 		}
 		$once = true;
 
-		$tours = apply_filters( 'tour_list', array() );
-		if ( empty( $tours ) ) {
+		$_tours = apply_filters( 'tour_list', array() );
+		if ( empty( $_tours ) ) {
 			return;
 		}
+		$tours = array_filter(
+			$_tours,
+			function ( $tour ) {
+				$restrict_url = $tour[0]['tour_restrict_url'];
+				$url_pattern  = '#/wp-admin/post\.php\?post=#';
+				if ( empty( $tour[0]['tour_restrict_url'] ) || preg_match( $url_pattern, $_SERVER['REQUEST_URI'] ) ) {
```

Comment:
We should rather use `is_admin()` here.
```suggestion
				if ( empty( $restrict_url ) || is_admin() ) {
```

Reply:
I think `is_admin()` would be too generic and would not allow tours created for the backend to be restricted to certain URLs in the admin backend.

Reply:
Oh, I see. What's the exact purpose of this regex then?

Reply:
The purpose of this regex is to ensure that the tour launcher is displayed on the "Create New Tour" page when "Add Steps" is clicked. 
I noticed that when I create a new tour and I enter a URL for it to be restricted to, then save. On clicking "Add Steps", the tour launcher which should be displayed at the bottom right of the screen isn't displayed and this is because  `$tours` array has been filtered to remove the tour I created, because it is only supposed to show on the page that has the restrict url I entered when creating it.

Reply:
Ok, so could we then do something like?
```suggestion
if ( empty( $restrict_url ) || ( is_admin() && $_GET['action'] === 'edit' && $_GET['post'] === $post_id ) ) {
```